File: example/src/main/scala/example/features/Enums.scala
package example.features

// Feature fixture: Scala 3 enums.
//   - Simple enums (no parameters) map to Dart `enum`.
//   - Parameterized/ADT enums map to a Dart sealed hierarchy (phase 2 of this
//     task, after the simple form is working).

enum Filter:
  case All, Active, Completed

enum Priority:
  case Low, Medium, High, Urgent

File: sart-facadegen/tool/sample/enums.dart
// Small fixture with a Dart enum.
library enums_sample;

enum Priority { low, medium, high, urgent }

enum Size { small, medium, large }

class Task {
  final String title;
  final Priority priority;
  final Size size;
  Task(this.title, this.priority, this.size);
}
